src/txn/txn_recover.c

```diff
@@ -629,6 +629,17 @@ __wt_txn_recover(WT_SESSION_IMPL *session)
      */
     WT_NOT_READ(metafile, NULL);
 
+    /*
+     * While we have the metadata cursor open, we should check whether the history store file exists
+     * or not. If it does not, then we should not apply rollback to stable to each table and we
+     * should not perform a checkpoint. This might happen if we're upgrading from an older version.
+     */
+    metac->set_key(metac, WT_HS_URI);
+    ret = metac->search(metac);
+    if (ret == WT_NOTFOUND)
+        hs_does_not_exist = true;
+    WT_ERR_NOTFOUND_OK(ret);
```

Comment:
Is this also a good place to remove the WiredTigerLAS file if it exists? Or is it too soon?
@tammybailey I believe you have a ticket mentioning removing the WiredTigerLAS file on startup in 4.4 (amongst other things) - have you started on that part of the work?

Reply:
I already made the change but it's like two lines so feel free to take it off my hands.

Reply:
I'll stop trying to feature creep and leave it as was.